spring-cloud-starter-stream-sink-mongodb/src/main/java/org/springframework/cloud/stream/app/mongodb/sink/MongodbSinkConfiguration.java

```diff
@@ -58,4 +67,31 @@ public MessageHandler mongoDbSinkMessageHandler() {
 		return mongoDbMessageHandler;
 	}
 
+
+	@Bean
+	@GlobalChannelInterceptor(patterns = Sink.INPUT)
+	public ChannelInterceptor bytesToStringChannelInterceptor() {
+		return new ChannelInterceptor() {
+
+			@Override
+			public Message<?> preSend(Message<?> message, MessageChannel channel) {
+				if (message.getPayload() instanceof byte[]) {
+					String contentType = message.getHeaders().containsKey(MessageHeaders.CONTENT_TYPE)
+							? message.getHeaders().get(MessageHeaders.CONTENT_TYPE).toString()
+							: BindingProperties.DEFAULT_CONTENT_TYPE.toString();
+					if (contentType.contains("text") ||
+							contentType.contains("json") ||
+							contentType.contains("x-spring-tuple")) {
```

Comment:
I wonder if we need to be more strict here and support only `json` content-type.
According the MongoDB docs it has to be some specific JSON format anyway: https://docs.mongodb.com/manual/reference/mongodb-extended-json/.

See JavaDocs of the `Document parse`:
```
 /**
    * Parses a string in MongoDB Extended JSON format to a {@code Document}
    *
    * @param json the JSON string
    * @return a corresponding {@code Document} object
    * @see org.bson.json.JsonReader
    * @mongodb.driver.manual reference/mongodb-extended-json/ MongoDB Extended JSON
    */
   public static Document parse(final String json)
```

But at the same time what to do with all other content-types: just reject as non-supported and enforce end-users to only rely on the default one and even don't let to reconfigure?...